play-json/jvm/src/main/scala/play/api/libs/json/BigNumberParser.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package play.api.libs.json
+
+object BigNumberParser {
+
+  def parseBigDecimal(input: String, settings: JsonParserSettings): java.math.BigDecimal = {
+
+    // There is a limit of how large the numbers can be since parsing extremely
+    // large numbers (think thousand of digits) and operating on the parsed values
+    // can potentially cause a DDoS.
+    if (input.length > settings.bigDecimalParseSettings.digitsLimit) {
+      throw new DigitLimitException
```

Comment:
I'm really not fond of throwing `Exception`, specially as that's mainly used to then be transformed as `JsResult`. Rather directly return `Either[String, T]`, and transform that as `JsResult` (or as `Exception` is jackson).